openedx/core/djangoapps/credit/serializers.py

```diff
@@ -32,3 +41,81 @@ class CreditCourseSerializer(serializers.ModelSerializer):
     class Meta(object):
         model = CreditCourse
         exclude = ('id',)
+
+
+class CreditProviderSerializer(serializers.ModelSerializer):
+    """ CreditProvider """
+    id = serializers.CharField(source='provider_id')  # pylint:disable=invalid-name
+    description = serializers.CharField(source='provider_description')
+    status_url = serializers.URLField(source='provider_status_url')
+    url = serializers.URLField(source='provider_url')
+
+    class Meta(object):
+        model = CreditProvider
+        fields = ('id', 'display_name', 'url', 'status_url', 'description', 'enable_integration',
+                  'fulfillment_instructions', 'thumbnail_url',)
+
+
+class CreditEligibilitySerializer(serializers.ModelSerializer):
+    """ CreditEligibility serializer. """
+    course_key = serializers.SerializerMethodField()
+
+    def get_course_key(self, obj):
+        """ Returns the course key associated with the course. """
+        return unicode(obj.course.course_key)
+
+    class Meta(object):
+        model = CreditEligibility
+        fields = ('username', 'course_key', 'deadline',)
+
+
+class CreditProviderCallbackSerializer(serializers.Serializer):  # pylint:disable=abstract-method
+    """
+    Serializer for input to the CreditProviderCallback view.
+
+    This is used solely for validating the input.
+    """
+    request_uuid = serializers.CharField(required=True)
+    status = serializers.ChoiceField(required=True, choices=CreditRequest.REQUEST_STATUS_CHOICES)
+    timestamp = serializers.IntegerField(required=True)
+    signature = serializers.CharField(required=True)
+
+    def __init__(self, **kwargs):
+        self.provider = kwargs.pop('provider', None)
+        super(CreditProviderCallbackSerializer, self).__init__(**kwargs)
+
+    def validate_timestamp(self, value):
+        """ Ensure the request has been received in a timely manner. """
+        dt = from_timestamp(value)  # pylint:disable=invalid-name
```

Comment:
can we just use a full name variable here instead of initials and get rid of this pylint comment